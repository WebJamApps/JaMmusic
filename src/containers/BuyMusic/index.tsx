import { Component } from 'react';
import Jsb from './BuyLinks/joshShermanBand';
import Jss from './BuyLinks/joshShermanSolo';
import Jsyt from './MediaLinks/joshShermanYoutube';
import Jssp from './MediaLinks/joshShermanSpotify';
import Jsam from './MediaLinks/joshShermanApple';
import commonUtils from '../../lib/utils';

export default class BuyMusic extends Component {
  commonUtils: { setTitleAndScroll: (pageTitle: string, width: number) => void };

  constructor(props: Record<string, unknown>) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount(): void { this.commonUtils.setTitleAndScroll('Buy Music', window.screen.width); }

  render(): JSX.Element {
    return (
      <div
        className="page-content"
        style={{
          paddingRight: '15px', maxWidth: '800px', margin: 'auto', borderRight: 'none',
        }}
      >
        <h2 style={{
          textAlign: 'center', margin: '20px', fontWeight: 'bold', marginBottom: '10px',
        }}
        >
          Buy from CD Baby
        </h2>
        <div className="row top-row-style">
          <Jsb />
          <Jss />
        </div>
        <Jssp />
        <Jsyt />
        <Jsam />
      </div>
    );
  }
}
