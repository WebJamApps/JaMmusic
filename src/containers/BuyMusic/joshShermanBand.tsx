import React from 'react';

const JoshShermanBand = () => (
  <div className="material-content elevation2" style={{ maxWidth: '4in', margin: 'auto' }}>
    <h4 style={{ textAlign: 'center' }}>Josh Sherman Band</h4>
    <div style={{ textAlign: 'center' }}>
      <a
        href="https://www.amazon.com/Live-Central-Florida-2001-Explicit/dp/B0015HGSEO/ref=ntt_mus_dp_dpt_1"
        target="_blank"
        rel="noopener noreferrer"
        style={{
          display: 'inline-block',
          background: 'url(https://m.media-amazon.com/images/I/81hDicJieTL._SS500_.jpg) 18px 2px no-repeat, '
            + 'url(http://content.cdbaby.com/img/links/link-artwork-cart-dark-buy-now.png) 0px 0px no-repeat',
          backgroundSize: '175px, 211px',
          height: '233px',
          width: '211px',
        }}
        title="Josh Sherman Band: live from central Florida, 2001 - 2005"
      >
        <span style={{ display: 'none' }}>Buy Josh Sherman Band</span>
      </a>
    </div>
  </div>
);

export default JoshShermanBand;
