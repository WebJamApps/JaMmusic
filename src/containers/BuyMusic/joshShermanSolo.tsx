import React from 'react';

const JoshShermanSolo = () => (
  <div className="material-content elevation2" style={{ margin: 'auto', maxWidth: '4in' }}>
    <h4 style={{ textAlign: 'center' }}>Josh Sherman Solo Acoustic</h4>
    <div style={{ textAlign: 'center' }}>
      <a
        href="https://www.amazon.com/Solo-Acoustic-Explicit-Josh-Sherman/dp/B0013XP8MI/ref=ntt_mus_dp_dpt_2"
        target="_blank"
        rel="noopener noreferrer"
        style={{
          textAlign: 'center',
          display: 'inline-block',
          background: 'url(https://m.media-amazon.com/images/I/81PLWgAmBEL._SS500_.jpg) 18px 2px no-repeat, '
          + 'url(http://content.cdbaby.com/img/links/link-artwork-cart-light-buy-now.png) 0px 0px no-repeat',
          backgroundSize: '175px, 211px',
          height: '233px',
          width: '211px',
        }}
        title="Josh Sherman: Solo Acoustic"
      >
        <span style={{ display: 'none' }}>Buy Solo Acoustic</span>
      </a>
    </div>
  </div>
);

export default JoshShermanSolo;
