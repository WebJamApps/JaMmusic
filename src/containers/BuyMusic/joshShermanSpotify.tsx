import React from 'react';

const JoshShermanSpotify = (): JSX.Element => (
  <div className="spotify">
    <hr/>
    <h5>Also On Spotify</h5>
    <div className="row">
      <div className="col" style={{ paddingRight: '0px' }}>
        <iframe
          src="https://open.spotify.com/embed/artist/5IvBs06z4RksIE1WvqLULs"
          width="312"
          height="436"
          title="soloAcoustic"
        />
      </div>
      <p style={{ fontSize: '1pt' }}>{' '}</p>
      <div className="col" style={{ paddingRight: '0px' }}>
        <iframe
          src="https://open.spotify.com/embed/artist/4XGcA7sSHYypVflLH48KCx"
          width="312"
          height="436"
          title="soloAcoustic"
        />
      </div>
    </div>
  </div>
);

export default JoshShermanSpotify;
