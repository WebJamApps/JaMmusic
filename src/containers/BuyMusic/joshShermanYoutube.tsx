import React from 'react';

const JoshShermanYoutube = ():JSX.Element => (
    <div className='youtube' style={{ marginBottom:'14px' }}>
        <hr/>
    <h5>Also On YouTube</h5>
    <div className="material-content elevation2" style={{ maxWidth: '4in', margin: 'auto', height: '2in' }}>
        <a href="https://youtube.com/playlist?list=OLAK5uy_nmxiOsu9BvdnwfHZ2w4ix3AG0POpsJPKA" 
        className='fab fa-youtube fa-1x' style={{ color:'#333333', marginTop:'18px' }}
        > <strong>Josh Sherman</strong><br/><i>Solo Acoustic</i></a>
        <br/>
        <a href="https://youtube.com/playlist?list=OLAK5uy_mtLf1U_oKuSgkFRcXYpo5E0OJWnvRt1TI" 
        className='fab fa-youtube fa-1x' style={{ color:'#333333', marginTop:'18px' }}
        > <strong>Josh Sherman Band</strong><br/><i>live from central Florida, 2001 - 2005</i></a>
        </div>
    </div>
);

export default JoshShermanYoutube;
