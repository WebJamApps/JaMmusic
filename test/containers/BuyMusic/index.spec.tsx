import BuyMusic from 'src/containers/BuyMusic';
import renderer from 'react-test-renderer';

describe('BuyMusic', () => {
  it('renders correctly', () => {
    const bm = renderer.create(<BuyMusic />).toJSON();
    expect(bm).toMatchSnapshot();
  });
});
