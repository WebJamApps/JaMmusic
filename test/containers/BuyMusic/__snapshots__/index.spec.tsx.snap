// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyMusic renders correctly 1`] = `
<div
  className="page-content"
  style={
    {
      "borderRight": "none",
      "margin": "auto",
      "maxWidth": "800px",
      "paddingRight": "15px",
    }
  }
>
  <h2
    style={
      {
        "fontWeight": "bold",
        "margin": "20px",
        "marginBottom": "10px",
        "textAlign": "center",
      }
    }
  >
    Buy from CD Baby
  </h2>
  <div
    className="row top-row-style"
  >
    <div
      className="col"
    >
      <div
        className="material-content elevation2"
        style={
          {
            "height": "3in",
            "margin": "auto",
            "maxWidth": "3in",
          }
        }
      >
        <h5
          style={
            {
              "textAlign": "center",
            }
          }
        >
          Josh Sherman Band
        </h5>
        <div
          style={
            {
              "textAlign": "center",
            }
          }
        >
          <a
            href="https://www.amazon.com/Live-Central-Florida-2001-Explicit/dp/B0015HGSEO/ref=ntt_mus_dp_dpt_1"
            rel="noopener noreferrer"
            style={
              {
                "background": "url(https://m.media-amazon.com/images/I/81hDicJieTL._SS500_.jpg) 18px 2px no-repeat, url(http://content.cdbaby.com/img/links/link-artwork-cart-dark-buy-now.png) 0px 0px no-repeat",
                "backgroundSize": "175px, 211px",
                "display": "inline-block",
                "height": "233px",
                "width": "211px",
              }
            }
            target="_blank"
            title="Josh Sherman Band: live from central Florida, 2001 - 2005"
          >
            <span
              style={
                {
                  "display": "none",
                }
              }
            >
              Buy Josh Sherman Band
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="material-content elevation2"
        style={
          {
            "height": "3in",
            "margin": "auto",
            "maxWidth": "3in",
          }
        }
      >
        <h5
          style={
            {
              "textAlign": "center",
            }
          }
        >
          Solo Acoustic
        </h5>
        <div
          style={
            {
              "textAlign": "center",
            }
          }
        >
          <a
            href="https://www.amazon.com/Solo-Acoustic-Explicit-Josh-Sherman/dp/B0013XP8MI/ref=ntt_mus_dp_dpt_2"
            rel="noopener noreferrer"
            style={
              {
                "background": "url(https://m.media-amazon.com/images/I/81PLWgAmBEL._SS500_.jpg) 18px 2px no-repeat, url(http://content.cdbaby.com/img/links/link-artwork-cart-light-buy-now.png) 0px 0px no-repeat",
                "backgroundSize": "175px, 211px",
                "display": "inline-block",
                "height": "233px",
                "textAlign": "center",
                "width": "211px",
              }
            }
            target="_blank"
            title="Josh Sherman: Solo Acoustic"
          >
            <span
              style={
                {
                  "display": "none",
                }
              }
            >
              Buy Solo Acoustic
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col"
    style={
      {
        "flexBasis": "0",
        "flexGrow": "1",
        "maxWidth": "100%",
        "msFlexPositive": "1",
        "msFlexPreferredSize": "0",
      }
    }
  >
    <div
      style={
        {
          "margin": "auto",
          "marginTop": "50px",
          "textAlign": "center",
        }
      }
    >
      <div
        className="spotify"
      >
        <h4
          style={
            {
              "marginBottom": "0px",
            }
          }
        >
          Also On Spotify
        </h4>
        <div
          className="row"
          style={
            {
              "display": "flex",
              "flexWrap": "wrap",
              "marginLeft": "-15px",
              "marginRight": "-15px",
            }
          }
        >
          <div
            className="col"
            style={
              {
                "flexBasis": "0",
                "flexGrow": "1",
                "maxWidth": "100%",
                "msFlexPositive": "1",
                "msFlexPreferredSize": "0",
              }
            }
          >
            <iframe
              height="436"
              src="https://open.spotify.com/embed/artist/5IvBs06z4RksIE1WvqLULs"
              title="soloAcoustic"
              width="312"
            />
          </div>
          <p
            style={
              {
                "fontSize": "1pt",
              }
            }
          >
             
          </p>
          <div
            className="col"
            style={
              {
                "flexBasis": "0",
                "flexGrow": "1",
                "maxWidth": "100%",
                "msFlexPositive": "1",
                "msFlexPreferredSize": "0",
              }
            }
          >
            <iframe
              height="436"
              src="https://open.spotify.com/embed/artist/4XGcA7sSHYypVflLH48KCx"
              title="soloAcoustic"
              width="312"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col"
  >
    <div
      style={
        {
          "textAlign": "center",
        }
      }
    >
      <h4
        style={
          {
            "marginBottom": "12px",
            "marginTop": "30px",
          }
        }
      >
        Also on YouTube
      </h4>
      <div
        className="youtube elevation3 top-row-style"
        style={
          {
            "marginBottom": "50px",
            "textAlign": "left",
          }
        }
      >
        <a
          className="fab fa-youtube fa-2x"
          href="https://youtube.com/playlist?list=OLAK5uy_nmxiOsu9BvdnwfHZ2w4ix3AG0POpsJPKA"
          style={
            {
              "color": "#333333",
              "margin": "20px",
              "marginLeft": "5%",
            }
          }
        >
           
          Josh Solo Acoustic
        </a>
        <a
          className="fab fa-youtube fa-2x"
          href="https://youtube.com/playlist?list=OLAK5uy_mtLf1U_oKuSgkFRcXYpo5E0OJWnvRt1TI"
          style={
            {
              "color": "#333333",
              "margin": "20px",
              "marginLeft": "5%",
            }
          }
        >
           
          Josh Sherman Band
        </a>
      </div>
    </div>
  </div>
  <div
    className="col"
  >
    <div
      style={
        {
          "textAlign": "center",
        }
      }
    >
      <h4
        style={
          {
            "marginBottom": "12px",
            "marginTop": "30px",
          }
        }
      >
        Also on Apple Music
      </h4>
      <div
        className="appleMusic elevation3 top-row-style"
        style={
          {
            "marginBottom": "50px",
            "textAlign": "center",
          }
        }
      >
        <a
          className="fab fa-apple fa-2x"
          href="https://music.apple.com/nz/artist/josh-sherman-band/id80070957"
          style={
            {
              "color": "#333333",
              "margin": "20px",
              "marginLeft": "5%",
            }
          }
        >
           
          Josh Sherman Band
        </a>
      </div>
    </div>
  </div>
</div>
`;
